std/didexchange/signature/signature.go

```diff
@@ -82,9 +82,17 @@ func verifySignature(cs *didexchange.ConnectionSignature, pipe *sec.Pipe) (c *di
 	}
 
 	timestamp := int64(binary.BigEndian.Uint64(data))
+	now := time.Now().Unix()
+	diff := now - timestamp
+	if diff < 0 || diff > connectionSigExpTime {
+		// try little endian - TODO: format missing from spec?
+		glog.Warningf("signature timestamp %s is invalid for big endian encoding, try little endian", time.Unix(timestamp, 0))
+		timestamp = int64(binary.LittleEndian.Uint64(data))
+		diff = now - timestamp
+	}
 
-	if time.Now().Unix()-timestamp > connectionSigExpTime {
-		glog.Error("timestamp too old, connection verify signature!")
+	if diff < 0 || diff > connectionSigExpTime {
```

Comment:
.. and if-statement B seems to be the same which makes code quite difficult to read in fast. Smallest change, we could use a small helper function to calc that boolean like `checkExpTime()` and should we then add a comment above to second if `// try again with LittleEdian..` Or should we make a whole new function that does this expiration calculation in a loop where Little and Big-endian are clearly stated and where we can see BigEndians is the preferred one, for example? I hope I made my point or didn't read the code wrong.

All in all I think that this is a bug in the .Net implementation. Or could we check what others do?